### PR TITLE
efs components added

### PR DESCRIPTION
**Pull request by samidbb**
## Describe your changes
Add resources to enable using AWS EFS (Amazon Elastic Filesystem) as backend for Persistent Volume Claims in Kubernetes Cluster
- Provision EFS filesystem (instance) for EKS cluster by default using recommended setting for performance and security
- Enable EKS add on for EFS support by default
- Deploy StorageClass

## Issue ticket number and link
https://github.com/dfds/cloudplatform/issues/3008

## Checklist before requesting a review
- [x] I have tested changes in my sandbox
- [x] I have added the needed changes in the `test/integration` folder to apply my changes in QA. [Read the guide on adding environment variables in QA](https://wiki.dfds.cloud/en/ce-private/atlantis/adding-env-vars)
- [x] I have rebased the code to master (or merged in the latest from master)

## Is it a new release?
- [x] Apply a release tag `release:(major|minor|patch)`, following semantic versioning in [this guide](https://semver.org/) or `norelease` if there is no changes to the Terraform code

### CODE REVIEW

**Comment by samidbb** [comment]
rename module efs-fs -> eks-storage-efs-fs ?

**Comment by SEQUOIIA** [comment]
Will try and test it in my sandbox cluster tomorrow

**Comment by SEQUOIIA** [comment]
Tested in my sandbox, applies without any issues. I created a test deployment using a pvc with the storageClass "csi-efs". Works just fine. I can see the pv in the EFS service in AWS. Good stuff @samidbb !

**Review by SEQUOIIA — APPROVED** [review]
Tested in https://github.com/dfds/infrastructure-modules/pull/1591#issuecomment-2435030491

**Comment by avnes** [comment]
I'll also have a look at this, because I want to learn it too :) 
But I am happy with Emil approving now, because I can proabbly not find time to test it until Monday or Tuesday

**Review by SEQUOIIA — DISMISSED** [review]

File: _sub/compute/eks-addons/main.tf [hunk 1]
```diff
@@ -144,3 +194,31 @@ resource "kubernetes_annotations" "gp2-not-default" {
...
+
+resource "kubernetes_storage_class" "csi-efs" {
+  metadata {
+    name = "csi-efs"
+    annotations = {
+      "storageclass.kubernetes.io/is-default-class" = "false"
+    }
+  }
+  storage_provisioner    = "efs.csi.aws.com"
+  reclaim_policy         = "Delete" # ?
```
Comment (avnes) [inline _sub/compute/eks-addons/main.tf:206]: I think the default should be Retain.

Comment (samidbb) [reply _sub/compute/eks-addons/main.tf:206]: @avnes why should this be different from what we offer through the EBS as we do here https://github.com/dfds/infrastructure-modules/blob/096983220481d55f2c15ee278e6f21ad46d9be80/_sub/compute/eks-addons/main.tf#L119?

Comment (avnes) [reply _sub/compute/eks-addons/main.tf:206]: OK, if it is the same, then in is OK

**Review by avnes — APPROVED** [review]

**Pull request merged**